docs/extend/api.md

```diff
@@ -12,348 +12,967 @@ To use the built-in REST API as part of an integration, see [Consuming the REST
+
+There are two types of relationships: `ToOne` and `ToMany`. You can use the `Schema\Relationship\ToOne` and `Schema\Relationship\ToMany` classes to define them.
 
 ```php
-use Flarum\Api\Event\WillSerializeData;
-use Flarum\Api\Controller\ListDiscussionsController;
-use Illuminate\Contracts\Events\Dispatcher;
+use Flarum\Api\Schema;
 
-return [
-    (new Extend\ApiController(ListDiscussionsController::class))
-        ->prepareDataForSerialization(function ($controller, $data, $request, $document) {
-            $data->load('myCustomRelation');
-        }),
-]
+public function fields(): array
+{
+    return [
+        Schema\Relationship\ToOne::make('user')
+            ->type('users'),
+        Schema\Relationship\ToMany::make('children')
+            ->type('labels'),
+    ];
+}
 ```
 
-## Serializers
+### Inclusion & Linkage
+
+You can mark a relationship as includable through the `includable` method. This means that the relationship can be included in the API response. You can also use the `withLinkage` and `withoutLinkage` methods to determine whether the relationship ID(s) should be included in the API response (`ToMany` relationships are not linked by default contrary to `ToOne` relationships).
 
-Before we can send our data to the frontend, we need to convert it to JSON:API format so that it can be consumed by the frontend.
-You should become familiar with the [JSON:API specification](https://jsonapi.org/format/).
-Flarum's JSON:API layer is powered by the [tobscure/json-api](https://github.com/tobscure/json-api) library.
+:::danger
 
-A serializer is just a class that converts some data (usually [Eloquent models](models.md#backend-models)) into JSON:API.
-Serializers serve as intermediaries between backend and frontend models: see the [model documentation](models.md) for more information.
-To define a new resource type, create a new serializer class extending `Flarum\Api\Serializer\AbstractSerializer`. You must specify a resource `$type` and implement the `getDefaultAttributes` method which accepts the model instance as its only argument:
+Adding linkage for `ToMany` relationships can lead to performance issues as it will include the IDs of all the related models in the API response. This is why it is not linked by default.
+
+:::
 
 ```php
-use Flarum\Api\Serializer\AbstractSerializer;
-use Flarum\Api\Serializer\UserSerializer;
+use Flarum\Api\Schema;
 
-class DiscussionSerializer extends AbstractSerializer
+public function fields(): array
 {
-    protected $type = 'discussions';
+    return [
+        Schema\Relationship\ToOne::make('user')
+            ->type('users')
+            ->includable()
+            ->withoutLinkage(),
+    ];
+}
+```
 
-    protected function getDefaultAttributes($discussion)
-    {
-        return [
-            'title' => $discussion->title,
-        ];
-    }
+### Polymorphic Relationships
+
+You use the `collection` method to define the resource types that a polymorphic relationship can point to.
```

Comment:
Are there docs by us or on the Eloquent site we can link to re: polymorphic relationships?

Reply:
good point